test/e2e/performanceprofile/functests/1_performance/irqbalance.go

```diff
@@ -0,0 +1,137 @@
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/mcps"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/nodes"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/pods"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/profiles"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/util"
+	"github.com/openshift/cluster-node-tuning-operator/test/framework"
+)
+
+var (
+	cs = framework.NewClientSet()
+)
+
+var _ = Describe("[performance] Checking IRQBalance settings", func() {
+	var workerRTNodes []corev1.Node
+	var profile *performancev2.PerformanceProfile
+	var performanceMCP string
+	var err error
+
+	BeforeEach(func() {
+		if discovery.Enabled() && testutils.ProfileNotFound {
+			Skip("Discovery mode enabled, performance profile not found")
+		}
+
+		workerRTNodes, err = nodes.GetByLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+		profile, err = profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+		performanceMCP, err = mcps.GetByProfile(profile)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Verify that worker and performance MCP have updated state equals to true
+		for _, mcpName := range []string{testutils.RoleWorker, performanceMCP} {
+			mcps.WaitForCondition(mcpName, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+		}
+	})
+
+	Context("Verify IRQ Balancing", func() {
+
+		var testpod *corev1.Pod
+
+		AfterEach(func() {
+			deleteTestPod(testpod)
+		})
+
+		It("Should not overwrite the banned CPU set on tuned restart", func() {
+			if profile.Status.RuntimeClass == nil {
+				Skip("runtime class not generated")
+			}
+			runtimeClass := profile.Status.RuntimeClass
+
+			irqLoadBalancingDisabled := profile.Spec.GloballyDisableIrqLoadBalancing != nil && *profile.Spec.GloballyDisableIrqLoadBalancing
+			if irqLoadBalancingDisabled {
+				Skip("this test needs dynamic IRQ balancing")
+			}
+
+			node := workerRTNodes[0] // any random node is fine
+			By(fmt.Sprintf("verifying worker node %q", node.Name))
+
+			var err error
+
+			bannedCPUs, err := nodes.BannedCPUs(node)
+			Expect(err).ToNot(HaveOccurred(), "failed to extract the banned CPUs from node %q", node.Name)
+			Expect(bannedCPUs.IsEmpty()).To(BeTrue(), "banned CPUs %v not empty on node %q", bannedCPUs, node.Name)
+
+			smpAffinitySet, err := nodes.GetDefaultSmpAffinitySet(&node)
+			Expect(err).ToNot(HaveOccurred(), "failed to get default smp affinity")
+
+			onlineCPUsSet, err := nodes.GetOnlineCPUsSet(&node)
+			Expect(err).ToNot(HaveOccurred(), "failed to get Online CPUs list")
+
+			Expect(smpAffinitySet.Equals(onlineCPUsSet)).To(BeTrue(), "found default_smp_affinity %v, expected %v", smpAffinitySet, onlineCPUsSet)
```

Comment:
I'm not sure I understand the test here.  Perhaps a comment would be helpful?

Reply:
right, added (and clarified a bit the error message)